spec/requests/purchases_spec.rb

```diff
@@ -0,0 +1,114 @@
+require 'rails_helper'
+
+RSpec.describe 'Purchases', type: :request do
+  let(:group) { Group.create(user: @user, name: 'Food', icon: 'missing_avatar.png') }
+  let(:purchase) { Purchase.create(name: 'Apples', amount: 5, author: @user) }
+  let(:group_purchase) { GroupPurchase.create(group: group, purchase: purchase) }
+  let(:valid_attributes) { { 'name' => 'Bananas', 'amount' => 5, 'author' => @user, 'group_ids' => [group.id] } }
+
+  before :each do
+    @user = User.create(name: 'Tom', email: 'tom@example.com', password: 'topsecret')
+    @user.confirm
```

Comment:
- [ ] Most of the tests within the files in the `spec/requests` folder, like this one is failing due to this `@user.confirm` line.  When I removed this line in all the `spec/requests` folder, all these 33 specs in the folder pass.

  ![image](https://github.com/katarighe/budget-app/assets/81468675/055750d7-6a4c-45e8-b2b0-74b703971bc9)